### ISSUE TITLE
New Xbox series X mapping for Linux after update

### ISSUE THREAD

**Issue by PKGaspi** [gabomdq/SDL_GameControllerDB#664]
After updating my Xbox series X controller's firmware, the mapping is broken for Linux. This is the new mapping:

```
030000005e040000120b00000f050000,Microsoft Xbox Series S|X Controller,a:b0,b:b1,y:b3,x:b2,start:b7,back:b6,leftstick:b9,rightstick:b10,leftshoulder:b4,rightshoulder:b5,dpup:b12,dpleft:b14,dpdown:b13,dpright:b15,leftx:a0,lefty:a1,rightx:a3,righty:a4,lefttrigger:a2,righttrigger:a5,platform:Linux
```

**Comment by offalynne** [comment]
7d416272d904e2d53eed353abda7468ba4fd023c

**Comment by 2000Slash** [comment]
Hey I have a quick question about this. (I don't really know if I should open a new issue). My Xbox Series X Controller with the exact same guid and the newest version still uses the hat input for the dpad instead of buttons, just like the other Xbox Series Controllers.
```
dpdown:h0.4,dpleft:h0.8,dpright:h0.2,dpup:h0.1
```
instead of
```
dpdown:b13,dpleft:b14,dpright:b15,dpup:b12
```
Since the entry for this controller was added, the dpad does not work for me in some programs (pcsx2). I use the xpad driver and face this issue on my main arch system and a fresh ubuntu install.

**Comment by 2000Slash** [comment]
Just tried it, but since xpad is the default driver for arch and I don't have another, it just doesn't get recognized anymore.

**Comment by offalynne** [comment]
Can you try with the *"fresh ubuntu install"*, please ?

**Comment by offalynne** [comment]
@PKGaspi can you provide any insight as to what your setup is like to compare? What distro you're on, what driver might be handling the Xbox pad, etc?

**Comment by 2000Slash** [comment]
I just tried in on a fresh ubuntu live and it is exactly the same as on my arch machine. The driver seems to be xpad. When I try to run pcsx2 the input is only accepted when I use the hat mappings. When I unload xpad the controller is not recognized anymore

**Comment by offalynne** [comment]
Last thing, can you please confirm either with an Xbox Console or on Windows with the Xbox Accessories app that your controller is on the latest firmware? If so please open a PR with the relevant change.

**Comment by 2000Slash** [comment]
Yup just tried to update it on my windows pc, but it says its on the newest version.

**Comment by PKGaspi** [comment]
> @PKGaspi can you provide any insight as to what your setup is like to compare? What distro you're on, what driver might be handling the Xbox pad, etc?

For testing I relied on my Arch system with the xpadneo driver. At the time, this was the only driver that supported bluetooth conection with the newest xbox controller. Doing further testing, I realized that the mappings are still wrong when the gamepad is used wireless with xpadneo, although they're great when cabled. 

After uninstalling xpadneo and restarting (using xpad I suppose but don't know how to check) the controller works fine in both cases and I can't reproduce the issue @2000Slash mentions.

I'm testing with the stable Godot 4.0.1 release. I have 2 controllers to test, updated the same day. Perhaps they're missing a newer update? I can check later for sure with the Xbox Accessories app.

**Comment by 2000Slash** [comment]
You can check if xpad is loaded with `lsmod | grep xpad`. If should show xpad on the left and then some other stuff. You can unload the module with `sudo rmmod xpad`. Then the controller stopped working for me. I used the controller with cable so when I plugged it back in the xpad driver reloaded and the controller inputs worked again (except the dpad).

**Comment by offalynne** [comment]
@2000Slash please see @PKGaspi's note. Try xpadneo and see how you get on.
